File: LeetCode/C#/1313. Decompress Run-Length Encoded List.cs
public class Solution {
    public int[] DecompressRLElist(int[] nums) {
        List<int> ans = new List<int>();
        
        for(var i = 0; i < nums.Length - 1; i += 2)
        {
            while(nums[i] != 0)
            {
                ans.Add(nums[i+1]);
                nums[i]--;
            }
        }
        
        return ans.ToArray();
    }
}

File: LeetCode/C++/1313. Decompress Run-Length Encoded List.cpp
class Solution
{
public:
    vector<int> decompressRLElist(vector<int> &nums)
    {
        vector<int> ans;

        for (int i = 0; i < nums.size(); i += 2)
        {
            int freq = nums[i];
            int val = nums[i + 1];

            while (freq > 0)
            {
                ans.push_back(val);
                freq--;
            }
        }

        return ans;
    }
};